packages/editor/src/plugins/audio/toolbar.tsx

```diff
@@ -26,6 +31,8 @@ export const AudioToolbar = ({
   setShowSettingsModal: Dispatch<SetStateAction<boolean>>
 }) => {
   const audioStrings = useEditorStrings().plugins.audio
+  const ref = useRef<HTMLButtonElement>(null)
+  const shadowRoot = useShadowRoot(ref)
```

Comment:
I don't understand why you pass the opener-button ref as the containerRef to `useShadowRoot`. Could you explain?
Also I'd love to move this complexity inside the modal instead of having it in all the consumers. Would that be possible somehow?

Reply:
Nice, happy about that!